Configuration/TypoScript/Config.txt
config {
    tx_mkforms {
        loadJsFramework = 1
        mayLoadScriptaculous = 0
        jsframework {
            jscore = jquery
            jscore {
                tx_mkforms_base = EXT:mkforms/Resources/Public/JavaScript/prototype/addons/base/Base.js
                basewrapper = EXT:mkforms/Resources/Public/JavaScript/wrapper.js
                wrapper = EXT:mkforms/Resources/Public/JavaScript/jquery/wrapper.js
            }
        }
    }
}


ChangeLog
0.6.5 - released 2017-05-22

Bug fixes:
- [#5798] Match only the first 4 characters when matching FE users by ANID
- [#5797] Update the ameos_formidable URL in the manual


0.6.4 - released 2017-04-12

Changes:
- [#5793] Allow oelib up to 1.9.99

Bug fixes:
- [#5792] Heating types are set even if "false" in OpenImmo


0.6.3 - released 2017-02-18

Features:
- [#5653] Add a configuration for PDF rendering of the manual

Bug fixes:
- [#5235] Explicitly provide the configuration check namespaces
- [#5420] Stop calling deprecated config check methods


0.6.2 - released 2015-12-04

Bug fixes:
- [#5386] Submitting the search keeps the current pagination
- [#5321] Update .htaccess to be Apache 2.4 compatible
- [#5362] Display the provision as a free-text field

Changes:
- [#5399] Document the new FORMidable in the upgrade notes


0.6.1 - released 2015-06-26

Bug fixes:
- [#5337] Crash when deleting and updating an object in the same import
- [#5336] Test crash in the RealtyMapper tests with an empty DB
- [#5333] Update the FORMidable download link
- [#5280] Download link to FORMidable is split across several lines
- [#5249] Missing comma in tca.php

Changes:
- [#5254] Document the Prototype/jQuery clash

Cleanup:
- [#5344] Clean up the OpenImmo import classes
- [#5205] Drop global variables from pi1_wizicon


0.6.0 - released 2015-01-17

Features:
- [#3570] Use cropscale for generating thumbnails
- [#4163] Add a decimal mark and thousands separator to the FE locallang
- [#5225] Add first name and salutation for objects
- [#5186] Add an automatic update function for contact_phone
- [#5027] Import the fields for the "Energiepass"/"Energieausweis"
- [#5027] Display the fields for the "Energiepass"/"Energieausweis"
- [#5027] Add the fields for the "Energiepass"/"Energieausweis"
- [#4678] Switch to Google Maps V3 (initial patch by Judith Haffner)
- [#3822] Use the Google geocoding V3 from oelib in realty
- [#4587] Add a field "Sea view" to the realty objects
- [#2671] Add model::isEmpty
- [#4413] Add a field "Distance to the sea" to the realty objects
- [#4118] Add some more area fields
- [#4090] Add additional image subparts in the single view
- [#4084] Add a configuration about which JavaScript libraries to include in the single view and image upload
- [#4103] Allow to upload a separate thumbnail image for the images
- [#4077] Provide a "floor" subpart & marker in the list view
- [#4089] Add an option to hide the rent and price in the list view and single view depending on the object status
- [#4087] Add a single view subpart for the status
- [#4100] Fill status markers in the list view
- [#4072] Replace the "rented" field with a "status" field
- [#4074] Add a field "position" to the images record
- [#4075] Add an "enableLightbox" TS setup switch
- [#4072] The clean-up script should also clean up PDFs
- [#4078] Allow to select the HTML template in the flexforms
- [#4012] Provide attached PDFs for download in the single view
- [#4016] Import attached documents (PDFs)
- [#4062] Add a sorting column to the documents
- [#4014] Add a Model and Mapper for related documents
- [#4000] Add a field "Bodenbelag" (flooring)
- [#3999] Add a field "Ausstattungs-Kategorie" (furnishing category)
- [#4017] Automatically make some content subtypes uncached
- [#3998] Add, import and display some fields
- [#4022] Add an icon for documents records to the BE
- [#2876] Make images sortable in the BE, thanks to Christian Stupp
- [#4011] Add a field for related documents (uploaded PDFs)
- [#3725] Validate the terms checkbox in the contact form
- [#3726] Include the checked checkboxes in the contact form e-mail
- [#3724] Add some new fields to the contact form
- [#3721] Add a TS setup configuration & flexforms for the FE page with the terms
- [#3722] Add some checkboxes for displaying additional fields in the contact form
- [#3602] In the contact data in the single view, display the new phone number fields
- [#3601] Import the two new phone number fields
- [#3600] Add the two new phone number fields to the FE editor
- [#3599] Add two phone number fields to the DB and TCA
- [#4162] Add a field "Mieteinnahmen (Soll)"

Localization:
- [#4606] Update the ES localization (thanks to Martin del Barco)
- [#3997] Change some field titles in the FE and BE

Bug fixes:
- [#5232] FE labels in 6.2 are in English instead of German
- [#5190] Fix code inspection warnings
- [#5224] Undefined variable $localeConvention in the filter form
- [#4343] Some unit tests read the real TS setup
- [#4319} Change != to <> in SQL
- [#5189] private field isTestMode is used in subclass
- [#5182] cHash is missing from many links
- [#5187] mb_strlen(): Unknown encoding in tx_realty_Model_RealtyObjectTest
- [#5094] Update script needs to update the number of images
- [#5173] PHP crash in the BE module if no import path is set
- [#5178] Use time() instead of mktime()
- [#5163] Make the insert record shortcut TS compatible with 6.2
- [#5175] Update Prototype, Scriptaculous and Lightbox
- [#5172] Fix core deprecation notices in TYPO3 CMS 6.2
- [#5174] Avoid feInterface deprecation warnings in 6.2
- [#5057] Test error in tx_realty_FrontEnd_MyObjectsListViewTest
- [#4696] Unit test crash about e-mail and caches in TYPO3 CMS 6.2
- [#5083] Failing test in tx_realty_FrontEnd_AbstractListViewTest, thanks to Nicole Cordes
- [#5063] Error about missing token when running the OpenImmo import from the BE module in 6.2
- [#5020] Error message: The extension CSS Styled Content is not loaded
- [#5008] Call loadTCA only for TYPO3 CMS < 6.1
- [#4572] Add .htaccess to all folders with private resources
- [#4648] Drop early __destruct calls
- [#4363] Skip the configuration check in the unit tests
- [#4847] Hide the images and documents in the BE lists
- [#4792] Error in tx_realty_Import_OpenImmoImportTest
- [#4779] Single view Google Maps is zoomed in too far
- [#4737] Missing currency sign for deposit and provision
- [#1927] Unit tests don't run properly on Windows

Configuration/RealUrl/Configuration.php
<?php

/**
 * This class adds a RealURL configuration.
 *
 * @author     Oliver Klee <typo3-coding@oliverklee.de>
 */
class Tx_Realty_Configuration_RealUrl_Configuration
{
    /**
     * Adds RealURL configuration.
     *
     * @param mixed[][] $parameters the RealUrl configuration to modify
     *
     * @return mixed[][] the modified RealURL configuration
     */
    public function addConfiguration(array $parameters)
    {
        $preVariables = [
            [
                'GETvar' => 'no_cache',
                'valueMap' => [
                    'no_cache' => 1,
                ],
                'noMatch' => 'bypass',
            ],
        ];

        $paginationGetVariable = [
            'GETvar' => 'tx_realty_pi1[pointer]',
            'valueMap' => [
                '1' => '0',
                '2' => '1',
                '3' => '2',
                '4' => '3',
                '5' => '4',
                '6' => '5',
                '7' => '6',
                '8' => '7',
                '9' => '8',
                '10' => '9',
                '11' => '10',
                '12' => '11',
                '13' => '12',
                '14' => '13',
                '15' => '14',
                '16' => '15',
                '17' => '16',
                '18' => '17',
                '19' => '18',
                '20' => '19',
                '21' => '20',
                '22' => '21',
                '23' => '22',
                '24' => '23',
                '25' => '24',
                '26' => '25',
                '27' => '26',
                '28' => '27',
                '29' => '28',
                '30' => '29',
                '31' => '30',
            ],
            'noMatch' => 'bypass',
        ];

        $realtyObjectGetVariable = [
            'GETvar' => 'tx_realty_pi1[showUid]',
            'lookUpTable' => [
                'table' => 'tx_realty_objects',
                'id_field' => 'uid',
                'alias_field' => 'title',
                'addWhereClause' => ' AND NOT deleted',
                'useUniqueCache' => true,
                'useUniqueCache_conf' => [
                    'strtolower' => 1,
                    'spaceCharacter' => '-',
                ],
                'autoUpdate' => true,
            ],
        ];

        $modeGetVariable = ['GETvar' => 'tx_realty_pi1[mode]', 'valueMap' => ['-' => '0']];
        $removeGetVariable = ['GETvar' => 'tx_realty_pi1[remove]', 'valueMap' => ['remove' => '1', '-' => '0']];
        $deleteGetVariable = ['GETvar' => 'tx_realty_pi1[delete]', 'valueMap' => ['remove' => '1', '-' => '0']];
        $ownerGetVariable = ['GETvar' => 'tx_realty_pi1[owner]', 'valueMap' => ['-' => '0']];
        $uidGetVariable = ['GETvar' => 'tx_realty_pi1[uid]', 'valueMap' => ['-' => '0']];

        return array_merge_recursive(
            $parameters['config'],
            [
                'preVars' => $preVariables,
                'postVarSets' => [
                    '_DEFAULT' => [
                        'immo' => [
                            $paginationGetVariable, $realtyObjectGetVariable, $modeGetVariable, $removeGetVariable,
                            $deleteGetVariable, $ownerGetVariable, $uidGetVariable,
                        ],
                    ],
                ],
            ]
        );
    }
}
